packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1100,6 +1117,32 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     return true;
   }
 
+  /// Immediately remove [route] and [Route.dispose] it.
+  ///
+  /// The route's animation does not run and the future returned from pushing
+  /// the route will not complete. Ongoing input gestures are cancelled. If
+  /// the [Navigator] has any [Navigator.observers], they will be notified with
+  /// [NavigatorObserver.didRemove].
+  ///
+  /// This method is used to dismiss dropdown menus that are up when the screen's
+  /// orientation changes.
+  void removeRoute(Route<dynamic> route) {
+    assert(route != null);
+    assert(!_debugLocked);
+    assert(() { _debugLocked = true; return true; });
+    assert(route._navigator == this);
+    final int routeIndex = _history.indexOf(route);
+    assert(routeIndex != -1);
+    setState(() {
+      _history.removeAt(routeIndex);
+      for (NavigatorObserver observer in widget.observers)
+        observer.didRemove(route, routeIndex > 0 ? _history[routeIndex - 1] : null);
```

Comment:
Yes, I think didChangeNext() and didChangePrevious() need to be called, good point.